src/facilities-context/reducer.tsx

```diff
@@ -0,0 +1,69 @@
+import { Facility } from "../page-multi-facility/types";
+import * as actions from "./actions";
+import { FacilitiesState } from "./FacilitiesContext";
+
+export function facilitiesReducer(
+  state: FacilitiesState,
+  action: actions.FacilitiesActions,
+): FacilitiesState {
+  switch (action.type) {
+    case actions.REQUEST_FACILITIES:
+      return Object.assign({}, state, { loading: true });
+
+    case actions.RECEIVE_FACILITIES:
+      return Object.assign({}, state, {
+        loading: false,
+        facilities: action.payload,
+      });
+
+    case actions.RECEIVE_FACILITIES_ERROR:
+      return Object.assign({}, state, {
+        loading: false,
+        failed: true,
+        facilities: {},
+      });
+
+    case actions.CREATE_OR_UPDATE_FACILITY: {
+      const facility = action.payload as Facility;
+      if (!facility?.id) return state;
```

Comment:
I think I put this in case the `saveFacility` failed, but that would be caught by the try/catch error in the function. I think I'll remove this check here and instead add a guard to check the facility in the `createOrUpdateFacility` function before dispatching.

i.e.:
```
       const updatedFacility = await saveFacility(scenarioId, facility);
        if (updatedFacility && updatedFacility.id) {
          dispatch({
            type: CREATE_OR_UPDATE_FACILITY,
            payload: { ...updatedFacility },
          });
        }
```